testcafe/searchField.test.ts

```diff
@@ -0,0 +1,81 @@
+import { Console } from "console";
+import { Selector } from "testcafe";
+import { Config } from "./testcafe.config";
+
+fixture`Search field test`.page`${Config.baseUrl}`
+    .beforeEach( async t => {
+        const nextButton = Selector("a").withAttribute("role", "button").child().withText("Weiter");
+        const websiteText = Selector("html").textContent;
+        const titleForIntro1 = "Willkommen bei der CovMap";
+        const titleForIntro2 = "Was ist die CovMap?";
+        const titleForIntro3 = "Was zeigt mir die CovMap an?";
+        const titleTextForPostalCode = "Für Dein regionales Risiko brauchen wir noch die Postleitzahl Deines Wohnortes";
+        
+        await t.expect(websiteText).contains(titleForIntro1).click(nextButton);
+        await t.expect(websiteText).contains(titleForIntro2).click(nextButton);
+        await t.expect(websiteText).contains(titleForIntro3).click(nextButton);
+        await t.expect(websiteText).contains(titleTextForPostalCode).pressKey("1 0 1 1 5 tab space tab tab enter");
+        await t.expect(websiteText).notContains(titleTextForPostalCode);
+    })
+    .afterEach( async t => {
+        // place holder
+    });
+
+test("find search field", async (t) => {
+    var searchField = Selector('#autocomplete').withAttribute("placeholder", "PLZ oder Landkreis")
+    await t.expect(searchField.exists).ok();
+    await t.click(searchField).typeText(searchField, "Stuttgart").pressKey("enter");
```

Comment:
Clicking and typing will automatically check if an element exists. I think we could remove the explicit check that it exists. (Also in other cases below.)

Reply:
I prefer to test if the components exists before doing any action on it. It makes debugging easier by making the granularity smaller.
note: I do not have lot of experience with testcafe and there error reporting, so my decisions are pretty conservative.

Reply:
I've made some test with testcafe, the error messages are clear enough.
existing and visibility tests could be rmoved.
![image](https://user-images.githubusercontent.com/9567496/97816228-1e0f7880-1c94-11eb-9dc0-15321d7d654e.png)